components/browser/domains/src/main/java/mozilla/components/browser/domains/DomainAutoCompleteProvider.kt

```diff
@@ -116,11 +117,14 @@ class DomainAutoCompleteProvider {
         this.useShippedDomains = useShippedDomains
 
         if (loadDomainsFromDisk) {
-            launch(UI) {
-                val domains = async(CommonPool) { Domains.load(context) }
-                val customDomains = async(CommonPool) { CustomDomains.load(context) }
-
-                onDomainsLoaded(domains.await(), customDomains.await())
+            CoroutineScope(Dispatchers.Main).launch {
+                // If a 'load' call throws, cancel the other one and re-throw.
+                coroutineScope {
```

Comment:
Those async calls are already part of the outer scope you create with `CoroutineScope()`. There shouldn't be any need to wrap them in another scope afaik?

Reply:
But this should work with the outer scope already?